server/src/omtDocumentInformationProvider.ts

```diff
@@ -0,0 +1,359 @@
+                                declaredImport: {
+                                    module: diMatchResult.declaredImportModule,
+                                }
+                            }));
+                    }
+                    else {
+                        fileImportsResult && calledObjects.push(...getReferencesToOtherFilesForCode(fileImportsResult.availableImports, l, line));
+                    }
+                }
+            }
+            else {
+                fileImportsResult && calledObjects.push(...getReferencesToOtherFilesForCode(fileImportsResult.availableImports, l, line));
+                fileImportsResult && calledObjects.push(...getLocalLocationsForCode(fileImportsResult.definedObjects, l, line));
+            }
+        }
+        console.timeEnd('getOmtDocumentInformation for ' + document.uri);
+        return {
+            documentLinks,
+            definedObjects: fileImportsResult?.definedObjects ?? [],
+            calledObjects,
+            availableImports: fileImportsResult?.availableImports ?? []
+        };
+    }
+}
+
+function getUriMatch(line: string, document: TextDocument, shorthands: Map<string, string>) {
+    const uriMatch = omtUriMatch.exec(line);
+    if (uriMatch) {
+        // match[0] is the full match inluding the whitespace of match[1]
+        // match[1] is the whitespace and optional quotes. both of which we don't want to include in the linked text
+        // match[2] is the link text, including the @shorthands
+        const link = replaceStart(uriMatch[2].trim(), shorthands);
+        const url = isAbsolute(link) ? resolve(document.uri, link) : toAbsolutePath(document, link);
+        return { uriMatch, url };
+    }
+}
+
+function getDiMatch(line: string) {
+    const diMatch = declaredImportMatch.exec(line);
+    if (diMatch) {
+        // match[0] is the full line match including the trailing colon
+        // match[1] is the whitespace prepending the import
+        // match[2] is the module name
+        const declaredImportModule = '' + diMatch[2];
+        return { diMatch, declaredImportModule }
+    }
+}
+
+function getReferencesToOtherFilesForCode(fileImports: OmtImport[], l: number, line: string): OmtLocalObject[] {
+    return findUsagesInLine(fileImports.map(x => x.name), l, line);
+}
+
+function getLocalLocationsForCode(declaredObjects: OmtLocalObject[], l: number, line: string): OmtLocalObject[] {
+    return findUsagesInLine(declaredObjects.map(x => x.name), l, line);
+}
+
+function findUsagesInLine(declaredObjects: string[], l: number, line: string): OmtLocalObject[] {
+    const documentLinks: OmtLocalObject[] = [];
+
+    declaredObjects.filter(declaredObject => line.includes(declaredObject)).forEach(declaredObject => {
+        const characterIndex = line.search(new RegExp(`${declaredObject}(?=[^a-zA-Z0-9]|$)`)) ?? line.indexOf(declaredObject);
```

Comment:
You'd probably want to check that the match begins at a word boundary or @ so that `someQuery` does not show up as a match for `bothersomeQuery`.

Reply:
I think I can use the same construction I use that the end of the string? So that the character before the declaredObject is not a letter, nor a number?